src/models/shared-model/shared-model.ts

```diff
@@ -57,17 +58,57 @@ export const SharedModel = types.model("SharedModel", {
             // shared model
             if (snapshot.id !== self.id) {
                 console.log("tried to apply shared model snapshot from different tree", {selfId: self.id, snapshot});
-                return;
+                return Promise.resolve();
```

Comment:
It isn't really an error if a snapshot is being applied from a different tree, there isn't any checking at the container level used to decide which shared model to send which snapshot too.  So I guess I make this a warning and configure eslint. That way we'll know to fix it if we copy this code into CLUE.

For the promise, I don't think I can make this async because this is a MST action. It should be run synchronously or be a generator inside of a flow.  This is something I need to understand better though I just did what works here without figuring that part out.

Reply:
You're right -- I wasn't thinking about the consequences of being in an action when I made my async/await suggestions.